.github/workflows/release.yml

```diff
@@ -3,41 +3,12 @@ name: Build and Release
 on:
   workflow_run:
     workflows: ["Test"]
-    branches:
-      - main
```

Comment:
Removing this would make releases happen every time a tag starting with "v" is pushed to any branch. Keeping in mind that this workflow runs from Test.

Reply:
Hmm you are correct, but it seems the release workflow isn't ever triggered because the tag itself triggers `Test` and is seen as the branch (not `main`). I think we'll need to rework this logic unless I'm missing something

Reply:
Mmm, that's true. A branch and a tag are kind of the same thing. You don't push a tag to a branch, you push a tag corresponding to a commit (corresponding potentially to multiple branches). I think as long as tags are protected, then it is safe to remove this main branch check as you did.

Reply:
Ahh yes haha thank you, well said.

> I think as long as tags are protected, then it is safe to remove this main branch check as you did.

Sounds good!

Reply:
```
on:
  pull_request:
  workflow_dispatch:
  push:
    branches:
    - main
    tags:
    - "v*"
  release:
    types: [published]
```

Is this an `and` or an `or`? I feel like is an or, so either if you push to main, or push a tag (from any branch) a release will be executed.

Reply:
Yes, it is an `or` which means that the RELEASING docs for Nile are incorrect—or at least misleading. Good call, we should definitely rectify this.

Reply:
> "Only releasable branch" is a real check?

no, that's only per our release process. this is contracts' CONTRIBUTING:

> (2) Checkout the branch to be released. This should be main except in the event of a hotfix. For hotfixes, checkout the latest release branch.

Reply:
what's the motivation to remove this line?

Reply:
> what's the motivation to remove this line?

Since only pushed `v*` tags trigger `Test`, the release workflow will only see the tag as the branch (not `main`) and thus not trigger when it should

Reply:
To further summarize: the branch directive in the `Test` workflow does not prevent tags pushed from other branches triggering the workflow; however, the branch directive in `Build and Release` _does_ prevent this (the branch being the tag in this process). By keeping the branch directive in the release workflow, it will only trigger if the `Test` workflow is first triggered by a commit pushed to `main` with a tag—this won't ever happen because the `Test` workflow is _only_ triggered by `v*` tags.

Two things:
1. Those with push permissions must be mindful when pushing a tag and push from `main` because that will ensure the latest commit gets tagged.
2. Maybe we can include a check in the `Test` workflow where the tagged commit must match the latest commit in `main`?

Reply:
Thanks for the summary @andrew-fleming, 2 sounds like a great idea to me.